### PR TITLE
Move json extension to strict mode

### PR DESCRIPTION
**Pull request by mjbvz**
Updates the json extension to use strict mode TypeScript

Main changes are annotation where values may be undefined

### CODE REVIEW

File: extensions/json/client/src/jsonMain.ts [hunk 1]
```diff
@@ -244,16 +246,16 @@ function getSettings(): Settings {
 	};
 
 	// merge global and folder settings. Qualify all file matches with the folder path.
-	let globalSettings = workspace.getConfiguration('json', null).get<JSONSchemaSettings[]>('schemas');
-	if (Array.isArray(globalSettings)) {
-		collectSchemaSettings(globalSettings, workspace.rootPath);
+	let globalSettings = workspace.getConfiguration('json').inspect<JSONSchemaSettings[]>('schemas');
```
Comment (mjbvz) [inline extensions/json/client/src/jsonMain.ts]: This part required changing the logic. Seems the current version of the api does allow null for the resource here

**Comment by aeschli** [comment]
@mjbvz Maybe I missed a discussion, but did we decide to use strict mode in our code? I rather keep things as they are. The code just got uglier, with ! at several places, and I don't think we win much. And while this is a simple case, it just asks for all dependencies to also go strict. And that's just a lot of work that we should plan first.

**Comment by mjbvz** [comment]
@aeschli This is something I am investigating for debt week. Strict mode can help catch common programming mistakes and generally encourages higher quality design. See the number of bugs we have around `Cannot read property of X` for example. 

This also seems to be the direction that TS is encouraging. We have been missing out on a lot of additional type checking features because we are not using strict mode and the longer we go, the more painful any eventual migration becomes and the more helpful tooling we keep missing

**Comment by mjbvz** [comment]
Closing as out of date.

@aeschli I strongly recommend you look into moving the json extension to strict mode during debt week. My comment notes why this is beneficial and, as my attempts to update this PR demonstrate, moving to strict mode becomes more and more painful as more time goes on. Feel free to revive this PR or start from scratch

**Pull request closed without merging**